inference/core/workflows/core_steps/models/foundation/google_vision_ocr/v1.py

```diff
@@ -0,0 +1,181 @@
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "Google Vision OCR",
+            "version": "v1",
+            "short_description": "Detect text in images using Google Vision API",
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "model",
+        },
+        protected_namespaces=(),
+    )
+    type: Literal["roboflow_core/google_vision_ocr@v1"]
+    image: Union[WorkflowImageSelector, StepOutputImageSelector]
+    ocr_type: Literal["text_detection", "ocr_text_detection"]
+    api_key: Union[WorkflowParameterSelector(kind=[STRING_KIND]), str] = Field(
+        description="Your Google Vision API key",
+        examples=["xxx-xxx", "$inputs.google_api_key"],
+        private=True,
+    )
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name="text", kind=[STRING_KIND]),
+            OutputDefinition(name="language", kind=[STRING_KIND]),
+            OutputDefinition(
+                name="predictions", kind=[OBJECT_DETECTION_PREDICTION_KIND]
+            ),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+class GoogleVisionOCRBlockV1(WorkflowBlock):
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    def run(
+        self,
+        image: WorkflowImageData,
+        ocr_type: Literal["text_detection", "ocr_text_detection"],
+        api_key: str,
+    ) -> BlockResult:
+        # Decide which type of OCR to use and make the request to Google Vision API
+        if ocr_type == "text_detection":
+            type = "TEXT_DETECTION"
+        elif ocr_type == "ocr_text_detection":
+            type = "DOCUMENT_TEXT_DETECTION"
+        else:
+            raise ValueError(f"Invalid ocr_type: {ocr_type}")
+
+        response = requests.post(
+            "https://vision.googleapis.com/v1/images:annotate",
+            params={"key": api_key},
+            json={
+                "requests": [
+                    {
+                        "image": {"content": image.base64_image},
+                        "features": [{"type": type}],
+                    }
+                ]
+            },
+        )
+
+        if response.status_code != 200:
+            raise RuntimeError(
+                f"Request to Google Cloud Vision API failed: {str(response.json())}"
+            )
+
+        result = response.json()["responses"][0]
+
+        # Check for image without text
+        if "textAnnotations" not in result or not result["textAnnotations"]:
+            return {
+                "text": "",
+                "language": "",
+                "predictions": sv.Detections.empty(),
+            }
+
+        # Extract predictions from the response
+        text = result["textAnnotations"][0]["description"]
```

Comment:
good navigation in output - I must admit it is not 100% clear at the first glance how to parse response 👍